### DISCUSSION TITLE
How to generate png on HPC

### DISCUSSION THREAD

**Discussion by dannypanduro** [neperfepx/neper#347] [Q&A] [answered]
Hi All,

I might be asking a lot of questions soon because I'm learning to run programs on an HPC (i.e., neper and FEPX). I appreciate any help and patience!

I'm not sure if this is a cluster issue, but when I use 'neper -V', I can't generate a png. Below is my code: 
`neper -V simulation.tess -datacellcol id -imageformat 'png' -print 250tess`
I'm also running this in a conda environment, and have installed pov ray and libpng. 

Any thoughts? 

Thank you,
Richard

**Comment by dannypanduro** [comment, marked as answer]
I'll also include the error message I get:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.1-15
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/panduro1/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -V loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] simulation.tess -datacellcol id -imageformat png
         -print 150tess
Info   : ---------------------------------------------------------------
Info   : Loading tessellation...
Info   :     [i] Parsing file `simulation.tess'...
Info   :     [i] Parsed file `simulation.tess'.
Info   : Reading data (poly, col)...
Info   : Printing image...
Info   :     [o] Writing file `150tess.pov'...
Info   :   - Printing tessellation...
Info   :     [o] Wrote file `150tess.pov'.
Info   :   - Generating png file (1200x900 pixels)...
Info   :     [o] Writing file `150tess.png'...
Error  :     > File `150tess.png' could not be generated!
```

**Reply by xshang93** [reply]
I think you didn't load pov-ray library. Load pov-ray and you should be fine.

Edit: Thought I'd explain this a bit more. This error code you posted, is it from your personal pc or the HPC? (asking because it's running on only 1 thread.) Either way, when you do `$ which povray`, what do you get? I'm assuming you'll get no povray in your environment.

**Reply by dannypanduro** [reply]
I think you're right! I don't think povray was installed in my environment. 

I am using an HPC. I tried installing Povray as a module, but the system doesn't have a TIFF library so I need to install one of those as well. Thank you!
